### PR TITLE
GH-42224: [Java] Fix Typo in TestAceroSubstraitConsumer Test Method

### PR DESCRIPTION
**Pull request by llama90**
### Rationale for this change

Fixes a typo causing a `NullPointerException` in the test method

### What changes are included in this PR?

- Updating typo to fix Unit Test

### Are these changes tested?

Yes

### Are there any user-facing changes?

No.

* GitHub Issue: #42224

### CODE REVIEW

**Comment by llama90** [comment]
To verify this PR, I think we need to merge this PR.

- https://github.com/apache/arrow/pull/42217

**Review by lidavidm — APPROVED** [review]
LGTM, please rebase and then we can test

File: java/dataset/src/test/java/org/apache/arrow/dataset/substrait/TestAceroSubstraitConsumer.java [hunk 1]
```diff
@@ -582,9 +582,7 @@ public void testRunExtendedExpressionsProjectAndFilter() throws Exception {
     // Expression ADD: id + 2
     // Expression CONCAT: name + '-' + name
     String binarySubstraitExpressionProject =
-        "Ch4IARIaL2Z1bmN0aW9uc19hcml0aG1ldGljLnlhbWwSERoPCAEaC2FkZDppM"
-            + "zJfaTMyEhQaEggCEAEaDGNvbmNhdDp2Y2hhchoxChoaGBoEKgIQASIIGgYSBAoCEgAiBhoECgIoAhoTYWRkX3R3b190b19jb2x1"
-            + "bW5fYRpGCi0aKwgBGgRiAhABIgoaCBIGCgQSAggBIgkaBwoFYgMgLSAiChoIEgYKBBICCAEaFWNvbmNhdF9jb2x1bW5fYV9hbmR"
+        "Ch4IARIaL2Z1bmN0aW9uc19hcml0aG1ldGljLnlhbWwSERoPCAEaC2FkZDppMzJfaTMyEhQaEggCEAEaDGNvbmNhdDp2Y2hhchoxChoaGBoEKgIQASIIGgYSBAoCEgAiBhoECgIoAhoTYWRkX3R3b190b19jb2x1bW5fYRpGCi0aKwgBGgRiAhABIgoaCBIGCgQSAggBIgkaBwoFYgMgLSAiChoIEgYKBBICCAEaFWNvbmNhdF9jb2x1bW5fYV9hbmR"
```
Comment (vibhatha) [inline java/dataset/src/test/java/org/apache/arrow/dataset/substrait/TestAceroSubstraitConsumer.java:585]: What's this diff?

**Comment by lidavidm** [comment]
@github-actions crossbow submit java-jars

File: java/dataset/src/test/java/org/apache/arrow/dataset/substrait/TestAceroSubstraitConsumer.java (hunk 1 above)
Comment (llama90) [reply java/dataset/src/test/java/org/apache/arrow/dataset/substrait/TestAceroSubstraitConsumer.java]: I rolled back this! It's typo.

**Comment by llama90** [comment]
@lidavidm I've just finished rebasing! You are so fast!

**Comment by lidavidm** [comment]
Github notifies me on every push ^_^

**Pull request merged**